fontbe/src/orchestration.rs

```diff
@@ -248,41 +248,39 @@ impl TupleBuilder {
     }
 }
 
-#[repr(u8)]
-#[derive(Copy, Clone, Debug, PartialEq)]
-pub enum LocaFormat {
-    Short = 0,
-    Long = 1,
+// work around orphan rules.
+//
+// FIXME: Clarify if there's a good reason not to treat glyf/loca as a single
+// entity, for the purpose of persistence? like a struct that contains both
+// tables, and from which the format can be retrieved
+//
+// this whole thing needs a rethink, but this gets us working
+#[derive(Clone, Copy, Serialize, Deserialize, PartialEq)]
+pub struct LocaFormatWrapper(u8);
+
+impl From<LocaFormat> for LocaFormatWrapper {
+    fn from(value: LocaFormat) -> Self {
+        LocaFormatWrapper(value as _)
+    }
 }
 
-impl LocaFormat {
-    pub fn new(loca: &[u32]) -> LocaFormat {
-        // https://github.com/fonttools/fonttools/blob/1c283756a5e39d69459eea80ed12792adc4922dd/Lib/fontTools/ttLib/tables/_l_o_c_a.py#L37
-        if loca.last().copied().unwrap_or_default() < 0x20000
-            && loca.iter().all(|offset| offset % 2 == 0)
-        {
+impl From<LocaFormatWrapper> for LocaFormat {
+    fn from(value: LocaFormatWrapper) -> Self {
+        if value.0 == 0 {
             LocaFormat::Short
         } else {
             LocaFormat::Long
         }
     }
 }
 
-impl Persistable for LocaFormat {
+impl Persistable for LocaFormatWrapper {
     fn read(from: &mut dyn Read) -> Self {
-        let mut buf = Vec::new();
-        from.read_to_end(&mut buf).unwrap();
-        match buf.first() {
-            Some(0) => LocaFormat::Short,
-            Some(1) => LocaFormat::Long,
-            _ => {
-                panic!("serialized LocaFormat is invalid")
-            }
-        }
+        bincode::deserialize_from(from).unwrap()
```

Comment:
just curious, why do we need to use bincode to (de)serialize the loca?

Reply:
This isn't loca itself, but the loca format flag; we serialize it separately so that we can serialize loca in its final representation, even though we might not have the head table yet.